### PR TITLE
Celeba changes

### PR DESCRIPTION
**Pull request by dribnet**
I've made some changes to the celeba classifer and vae experiment, some of which might be generally useful. I'm posting all of the changes here for discussion.  There is some tension here between having a small and easy to understand example to support the paper and having a general purpose tool for further experiments. So I'm also happy to maintain this functionality in my own repo, perhaps split out in separate files.

That said, here's the changes in these commits:
- Added a bunch of command line options for things that were hard coded like
  - --classifier to use a different classifier filename
  - --model to use a different model filename
  - --batch-size to use different batch-sizes (generally smaller for memory reasons)
  - --z-dim to change network architecture allowing different sized latent space
  - --monitor-every and --checkpoint-every to change frequency of those events
- Split out `discriminative_term` in the cost function so it could be monitored separately (9565843)
- Ability to scale relative cost of reconstruction, kl, and discriminative (d9afe52)
- Classifier can train on a subset of labels given at runtime (dee459a)
- Can train against compatible fuel datasets other than CelebA (779d587)

Happy to discuss any of these that might be useful to clean up into a separate pull request.

### CODE REVIEW

File: experiments/train_celeba_vae.py [hunk 1]
```diff
@@ -307,8 +307,12 @@ def create_computation_graph():
             tensor.log(2 * pi) + 2 * log_sigma +
             (x - mu_theta) ** 2 / tensor.exp(2 * log_sigma)
         ).sum(axis=[1, 2, 3])
-        total_reconstruction_term = reconstruction_term
 
+        # FIXME: this is maybe the dumbest way to make [0.0] * batch_size ...
```
Comment (vdumoulin) [inline experiments/train_celeba_vae.py:311]: You could replace this with `tensor.zeros_like(kl_term)`.

**Comment by vdumoulin** [comment]
Here's what I'm okay with:
- Classifier path and save path flexibility
- Batch size flexibility
- Extension frequency flexibility
- Finer grained monitoring

However, I'm reluctant to modify the experiment itself. I feel like such modifications should go in a separate experiment file.

I would be okay with you adding another experiment file that acts more like a general purpose tool if you want.

**Comment by dribnet** [comment]
Great - I'll prepare that. I also like the idea of adding a general purpose tool alongside the more streamlined versions that support the paper.

**Comment by vdumoulin** [comment]
I'll close this PR and review the new one.

**Pull request closed without merging**